Upstream: WIP https://github.com/lupoDharkael/flameshot/pull/609

From 43c638d1f17ecd6f4c3e6b2f96fc2b549892914d Mon Sep 17 00:00:00 2001
From: Alexander Kapshuna <kapsh@kap.sh>
Date: Mon, 16 Sep 2019 02:12:17 +0300
Subject: [PATCH] Add zsh completion support

Both bash and zsh completions could be supported this way.
---
 docs/{bash-completion => shell-completion}/flameshot | 11 ++++++++++-
 flameshot.pro                                        | 10 +++++++---
 2 files changed, 17 insertions(+), 4 deletions(-)
 rename docs/{bash-completion => shell-completion}/flameshot (89%)

diff --git a/docs/bash-completion/flameshot b/docs/shell-completion/flameshot
similarity index 89%
rename from docs/bash-completion/flameshot
rename to docs/shell-completion/flameshot
index 58f5233..60c197e 100644
--- a/docs/bash-completion/flameshot
+++ b/docs/shell-completion/flameshot
@@ -1,5 +1,8 @@
-# bash-completion for flameshot command
+#compdef flameshot
+
+# Shell completion for flameshot command
 # To be installed in "/usr/share/bash-completion/completions/flameshot"
+# and "/usr/share/zsh/site-functions/"

 _flameshot() {
 	local prev cur cmd gui_opts full_opts config_opts
@@ -64,4 +67,10 @@ _flameshot() {
 			;;
 	esac
 }
+
+if [[ -n ${ZSH_VERSION} ]]; then
+    autoload -U bashcompinit
+    bashcompinit
+fi
+
 complete -F _flameshot flameshot
